internal/controller/volsync/vshandler.go

```diff
@@ -1113,6 +1119,44 @@ func (v *VSHandler) DeleteRD(pvcName string, pvcNamespace string) error {
 	return nil
 }
 
+func (v *VSHandler) pruneOldSnapshots(pvcNamespace string) error {
+	snapList := &snapv1.VolumeSnapshotList{}
+
+	err := v.listByOwner(snapList, pvcNamespace)
+	if err != nil {
+		return err
+	}
+
+	if len(snapList.Items) <= 1 {
+		return nil
+	}
+
+	// Sort snapshots by CreationTimestamp (ascending: oldest first)
+	slices.SortFunc(snapList.Items, func(a, b snapv1.VolumeSnapshot) int {
+		if a.CreationTimestamp.Before(&b.CreationTimestamp) {
+			return -1
+		}
+
+		return 1
+	})
+
+	for i := 0; i < len(snapList.Items)-1; i++ {
```

Comment:
(nit) It would be nice to extract this block and the one below into a reusable utility function that deletes a given list of objects. We are deleting a list of snapshots in two different places, one with `n-1` items and the other with `n` items.
https://github.com/RamenDR/ramen/blob/ca44e13fc020da35c9d4d5d32c8da6492e1fa9f0/internal/controller/volsync/vshandler.go#L1125-L1137